lib/timers.js

```diff
@@ -409,6 +393,26 @@ function processImmediate() {
   }
 }
 
+function tryOnImmediate(immediate, queue) {
+  var threw = true;
+  try {
+    immediate._onImmediate();
+    threw = false;
+  } finally {
+    if (threw) {
```

Comment:
Ideally you'd check `if (threw === true) {` here so V8's full-codegen doesn't generate an IC but emits an immediate boolean check.